### PR TITLE
Update labs to use NUCLEO-F303K8

### PR DESCRIPTION
**Pull request by dragonlock2**
- Updated lab 1 documentation to use F303K8
- Updated solution code and template code to use F303K8

### CODE REVIEW

File: lab1.md [hunk 1]
```diff
@@ -17,57 +17,80 @@ Objectives
...
+
+We've also standardized on using the [PlatformIO](https://platformio.org) IDE and [Mbed](https://os.mbed.com) framework to provide a streamlined development process and nicer API to control the underlying hardware.
 
 This lab will walk through some simple examples to get started.
 
+## Hardware Setup
+
+Before we get to coding, we need to setup the NUCLEO-F303K8 on a breadboard and connect an RGB LED and button to it. The pinout for the board can be found [here](https://os.mbed.com/platforms/ST-Nucleo-F303K8/). Breadboards are connected like in the following picture.
+
+![Image](docs/breadboard.jpg?raw=true)
```
Comment (ducky64) [inline lab1.md:30]: Might be useful to show the circuit diagram, and the breadboard diagram.

From the [ancient training documents](https://github.com/CalSol/training-docs), we had pictures that looked like this:
image
I can make updated diagrams - and I think diagrams might be more intuitive than a long text list? Feel free to leave a TODO here in this case.

File: lab1.md [hunk 2]
```diff
@@ -17,57 +17,80 @@ Objectives
...
 ## Sanity Check
-If you want to sanity-check your hardware and build environment, you can build and flash test code with this command (run this outside the `src` folder, **inside** the `braintrain` folder, while you have the programmer and board connected)
+If you want to sanity-check your hardware and build environment, you can build and flash test code with this command (run this outside the `src` folder, **inside** the `braintrain` folder, while you have the programmer and board connected).
 ```
 pio run -e solution_1_5 -t upload
 ```
 
+Of course, you can also just use PlatformIO's GUI.
+
 This should fade the RGB LED through all the colors over a period of 3 seconds.
```
Comment (ducky64) [inline lab1.md:54]: TODO we should also add something with the button

File: lab1.md [hunk 3]
```diff
@@ -17,57 +17,80 @@ Objectives
...
 
 ## Lab 1.1: Getting started
```
Comment (ducky64) [inline lab1.md:56]: Consider moving this to the pre-lab (especially if the campus wifi is still terribad), and consider moving the pairing up blurb to the actual lab part?

File: lab1.md [hunk 4]
```diff
@@ -17,57 +17,80 @@ Objectives
...
 ```c++
 #include "mbed.h"
 
 DigitalOut led1(LED1);
 
+RawSerial serial(SERIAL_TX, SERIAL_RX, 115200);
+
 // Replace pin assignments if different
-PwmOut ledR(D10);
+PwmOut ledR(D9);
```
Comment (ducky64) [inline lab1.md:76]: Is D9 to avoid using the CAN pin D10?

File: lab1.md [hunk 5]
```diff
@@ -17,57 +17,80 @@ Objectives
...
+A microcontroller is a full computer system on a chip, including a CPU, flash, RAM, and IO. You'll be working with the [NUCLEO-F303K8](https://os.mbed.com/platforms/ST-Nucleo-F303K8/) containing an STM32F303K8 that has a (up to) 72 MHz Cortex-M4F processor, 64 KiB of flash memory, 12 KiB of RAM, and various IO.
+
+We've also standardized on using the [PlatformIO](https://platformio.org) IDE and [Mbed](https://os.mbed.com) framework to provide a streamlined development process and nicer API to control the underlying hardware.
 
 This lab will walk through some simple examples to get started.
 
+## Hardware Setup
+
+Before we get to coding, we need to setup the NUCLEO-F303K8 on a breadboard and connect an RGB LED and button to it. The pinout for the board can be found [here](https://os.mbed.com/platforms/ST-Nucleo-F303K8/). Breadboards are connected like in the following picture.
```
Comment (ducky64) [inline lab1.md:28]: Perhaps inline embed the [F303K8 pinout image](https://os.mbed.com/media/uploads/bcostm/nucleo_f303k8_2017_10_10.png) (link only - do not commit the image to this repo unless we're sure about the rights)?

**Review by ducky64 — CHANGES_REQUESTED** [review]
Mostly looks good, some inline comments. Thanks for all the copy-editing!

Other issues:
- Where'd the breadboard file come from? Do we have the rights to use it and license it under the repository-wide license? We also might have a breadboard diagram from other training materials.
- Do we want to (and can we) write the text so it doesn't refer to any dev board, so if we end up doing this again, we'll have less changes?
- Do we still notice lag with floating point? The F303K8 has a FPU, doesn't it? If the floating point overhead is insignificant, I'm not sure how we want to write this section - perhaps disable the FPU in the compiler flags, slow down the processor clock, or get rid of it entirely (not preferred - this is an embedded gotcha)?

File: lab1.md (hunk 1 above)
Comment (dragonlock2) [reply lab1.md]: diagrams are added

File: lab1.md (hunk 4 above)
Comment (dragonlock2) [reply lab1.md:74]: yup

File: lab1.md (hunk 5 above)
Comment (dragonlock2) [reply lab1.md]: embedded

**Comment by dragonlock2** [comment]
- breadboard picture removed
- definitely something to do in the future
- switching to f042k6 so no longer relevant (did do a test on the f303k7 earlier and appeared that fpu was being used)

File: lab1.md (hunk 2 above)
Comment (dragonlock2) [reply lab1.md]: added a sanity check code to more easily tell if its working

**Review by ducky64 — APPROVED** [review]

**Pull request merged**